src/test_workflow/smoke_test/smoke_test_runner_opensearch.py

```diff
@@ -0,0 +1,81 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+from pathlib import Path
+from typing import Any
+
+import requests
+from openapi_core import Spec, validate_request, validate_response
+from openapi_core.contrib.requests import RequestsOpenAPIRequest, RequestsOpenAPIResponse
+
+from manifests.test_manifest import TestManifest
+from test_workflow.smoke_test.smoke_test_runner import SmokeTestRunner
+from test_workflow.test_args import TestArgs
+from test_workflow.test_result.test_component_results import TestComponentResults
+from test_workflow.test_result.test_result import TestResult
+from test_workflow.test_result.test_suite_results import TestSuiteResults
+
+
+class SmokeTestRunnerOpenSearch(SmokeTestRunner):
+
+    def __init__(self, args: TestArgs, test_manifest: TestManifest) -> None:
+        super().__init__(args, test_manifest)
+        logging.info("Entering Smoke test for OpenSearch Bundle.")
+
+        # TODO: Download the spec from https://github.com/opensearch-project/opensearch-api-specification/releases/download/main-latest/opensearch-openapi.yaml
+        spec_file = os.path.join(os.path.dirname(os.path.abspath(__file__)), "smoke_tests_spec", "opensearch-openapi.yaml")
+        self.spec_ = Spec.from_file_path(spec_file)
+        self.mimetype = {
+            "Content-Type": "application/json"
+        }
+        # self.openapi = openapi_core.OpenAPI.from_file_path(spec_file)
+
+    def validate_request_swagger(self, request: Any) -> None:
+        request = RequestsOpenAPIRequest(request)
+        validate_request(request=request, spec=self.spec_)
+        logging.info("Request is validated.")
+
+    def validate_response_swagger(self, response: Any) -> None:
+        request = RequestsOpenAPIRequest(response.request)
```

Comment:
why not validate request as well?

Reply:
There seem to be some issue on the tool about request validation as the string types are different.

Reply:
Can you add more details on it? Is there an issue on openapi repo, may be reference that.

Reply:
Looks like we need to parse the response as per the comments in the issue. How about JSON parsing with response string as an input?

Reply:
That could be too specific and it may not work for overall. That example is just one cases and is also restricted to the validation tool we are using.